Prevent accessing @driver unless set

tomhicks-bsf opened the pull request:
When there are zero steps in a file (as can happen when excluding
certain tags in a test run) it attempts to close the driver, having
never created it in the first place.

tomhicks-bsf committed:
Prevent accessing @driver unless set

When there are zero steps in a file (as can happen when excluding
certain tags in a test run) it attempts to close the driver, having
never created it in the first place.

samccone commented on src/support/index.coffee:
```diff
@@ -105,7 +105,10 @@ module.exports = ->
+      @driver.close()
+      @driver.quit()
+    else
+      $.defer()
```
shouldn't we return just a resolved promise here instead of a defered object that is in a non resolved state?

tomhicks-bsf replied:
Probably mate yeah. This was just any old crap that worked in five minutes. Just to raise the issue more than anything - was in a rush!

Cheers,Tom

On Mon, Apr 27, 2015 at 10:14 PM, Sam Saccone notifications@github.com
wrote:

> > @@ -105,7 +105,10 @@ module.exports = ->
> >      argv['prevent-browser-reload']?
> > 
> >    terminateDriver = =>
> > -    @driver.close()
> > -    @driver.quit()
> > -    if @driver?
> > -      @driver.close()
> > -      @driver.quit()
> > -    else
> > -      $.defer()
> >   shouldn't we return just a resolved promise here instead of a defered object that is in a non resolved state? 
> >   ---
> >   Reply to this email directly or view it on GitHub:
> >   https://github.com/mojotech/pioneer/pull/319/files#r29192054

tomhicks-bsf replied:
The other way is to check the existence of driver before calling terminate. Probably cleaner.

Cheers,Tom

On Mon, Apr 27, 2015 at 10:14 PM, Sam Saccone notifications@github.com
wrote:

> > @@ -105,7 +105,10 @@ module.exports = ->
> >      argv['prevent-browser-reload']?
> > 
> >    terminateDriver = =>
> > -    @driver.close()
> > -    @driver.quit()
> > -    if @driver?
> > -      @driver.close()
> > -      @driver.quit()
> > -    else
> > -      $.defer()
> >   shouldn't we return just a resolved promise here instead of a defered object that is in a non resolved state? 
> >   ---
> >   Reply to this email directly or view it on GitHub:
> >   https://github.com/mojotech/pioneer/pull/319/files#r29192054

samccone replied:
yeah I think this the idea behind this is a good PR. mind just updating a touch and we can get this merged in?

tomhicks-bsf replied:
Yeh man I'll sort it tomorrow. Hate depending on our own forks!

Cheers,Tom

On Mon, Apr 27, 2015 at 10:40 PM, Sam Saccone notifications@github.com
wrote:

> > @@ -105,7 +105,10 @@ module.exports = ->
> >      argv['prevent-browser-reload']?
> > 
> >    terminateDriver = =>
> > -    @driver.close()
> > -    @driver.quit()
> > -    if @driver?
> > -      @driver.close()
> > -      @driver.quit()
> > -    else
> > -      $.defer()
> >   yeah I think this the idea behind this is a good PR. mind just updating a touch and we can get this merged in?
> >   ---
> >   Reply to this email directly or view it on GitHub:
> >   https://github.com/mojotech/pioneer/pull/319/files#r29194433

samccone replied:
:+1: no prob thanks for the awesome work ps

tomhicks-bsf committed:
Use old version of webdriver

For some reason, certain commands are not failing
leaving the scenario running for ever, causing our
test runs to hang. The old version of selenium does
seemingly not have this problem.

tomhicks-bsf committed:
Merge remote-tracking branch 'upstream/master'

tomhicks-bsf committed:
Support cucumber AND/OR tags properly

Multiple tags should be supplied in multiple --tags arguments,
which allows logical ORs and ANDs to be applied to tags. This
creates --tags arguments according to the cucumber docs:

https://github.com/cucumber/cucumber/wiki/Tags

tags: ['@one', '~@two'] will run scenarios with tag @one, but not
scenarios with tag @two.

ThomWright committed:
Add glob support for require config paths

Fixes mojotech/pioneer#202

ThomWright committed:
New shrinkwrap for updated dependencies

tomhicks-bsf committed:
Update inflection dependency

0.3.2 was accidentally unpublished from npm can cannot be re added!

ThomWright committed:
Remove shrinkwrap

Probably gives no benefit
We're moving to node6, and the fixed dependencies might be causing issues

ThomWright committed:
Revert "Remove shrinkwrap"

This reverts commit 26eebc64a85413d7da9160adc13c318686217078.

Sticking with node 0.12 for now, so put the shrinkwrap back

Commit:
Move this fork to @mft npm org

Commit:
1.0.0

Commit:
Make webdriveer a peer dep
